astroid/context.py

```diff
@@ -42,7 +42,11 @@ class InferenceContext:
 
     max_inferred = 100
 
-    def __init__(self, path=None, nodes_inferred=None):
+    def __init__(
+        self,
+        path: set[tuple[NodeNG, str | None]] | None = None,
+        nodes_inferred: list[int] | None = None,
+    ):
```

Comment:
```suggestion
    ) -> None:
```

Reply:
Isn't `__init__` exempt from return typing?

Reply:
Ah of course. Just force of habit for me, feel free to ignore and resolve!